lib/forspell/runner.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require_relative 'reader'
+require_relative 'loaders/file_loader'
+
+module Forspell
+  class Runner
+    attr_reader :result, :total_errors
+
+    def initialize files:, speller:, reporter:
+      @files = files
+      @speller = speller
+      @reporter = reporter
+    end
+
+    def call
+      parsing_errors = []
+
+      @files.each do |path|
+        @reporter.file(path)
+        reader = Reader.new.for(path)
+        words = reader.read
+        errors = reader.parsing_errors || []
+        parsing_errors += errors unless errors.empty?
+        words.each do |word|
+          @reporter.error(word, @speller.suggest(word.text)) unless @speller.correct?(word.text)
+        end
+      end
+
+      @reporter.report(parsing_errors, @files.size)
```

Comment:
BTW, you probably don't need to gather ALL the errors in a variable. Maybe this would make sense (considering "one file processing method" above):
```ruby
# when we just start processing a file:
@reporter.file(path) # It remembers current file is "path", can gather stats "how many processed", can output "Processing..."
# ...on error
@reporter.parsing_error(e) # It can process it as "for file {current_file}, we had an error {e}"
```

Reply:
It would be OK for "on-the-fly" output - readable format, but I need all the errors for dictionary/JSON/YAML formats. Or not?

Reply:
But reporter could gather them (and format)!
Though, in fact, I believe that (at least for version 1) the following is the most reasonable:
1. The "normal" output (spelling errors) goes to STDOUT, in format requested
2. The "problems" output (that file can not be parsed) goes to STDERR, always in the same format.
This way, one can:
```
$ forspell ..... -fyaml > errors.yaml
```
↑ and have `errors.yaml` with spelling problems, and "that file can not be parsed" in console.